modules/simple_forms_api/app/services/simple_forms_api/form_upload_notification_email.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module SimpleFormsApi
+  class FormUploadNotificationEmail
+    attr_reader :form_number, :form_name, :first_name, :email, :date_submitted, :confirmation_number,
+                :lighthouse_updated_at, :notification_type
+
+    template_root = Settings.vanotify.services.va_gov.template_id
+    TEMPLATE_IDS = {
+      confirmation: template_root.form_upload_confirmation_email,
+      error: template_root.form_upload_error_email,
+      received: template_root.form_upload_received_email
+    }.freeze
+
+    SUPPORTED_FORMS = %w[21-0779 21-509 21P-0518-1 21P-0516-1].freeze
+
+    def initialize(config, notification_type:)
+      @notification_type = notification_type
+
+      check_missing_keys(config)
+      check_if_form_is_supported(config)
+
+      @form_number = config[:form_number]
+      @form_name = config[:form_name]
+      @first_name = config[:first_name]
+      @email = config[:email]
+      @date_submitted = config[:date_submitted]
+      @confirmation_number = config[:confirmation_number]
+      @lighthouse_updated_at = config[:lighthouse_updated_at]
+    end
+
+    def send(at: nil)
+      template_id = TEMPLATE_IDS[notification_type]
+      return unless template_id
+
+      sent_to_va_notify = if at
+                            enqueue_email(at, template_id)
+                          else
+                            send_email_now(template_id)
+                          end
+      StatsD.increment('silent_failure', tags: statsd_tags) if error_notification? && !sent_to_va_notify
+    end
+
+    private
+
+    def check_missing_keys(config)
+      all_keys = %i[form_number form_name first_name email date_submitted confirmation_number]
+
+      missing_keys = all_keys.select { |key| config[key].nil? || config[key].to_s.strip.empty? }
+
+      if missing_keys.any?
+        StatsD.increment('silent_failure', tags: statsd_tags) if error_notification?
+        raise ArgumentError, "Missing keys: #{missing_keys.join(', ')}"
+      end
+    end
+
+    def check_if_form_is_supported(config)
+      unless SUPPORTED_FORMS.include?(config[:form_number])
+        StatsD.increment('silent_failure', tags: statsd_tags) if error_notification?
+        raise ArgumentError, "Attempted to upload unsupported form: given form number was #{config[:form_number]}"
+      end
+    end
+
+    def send_email_now(template_id)
+      VANotify::EmailJob.perform_async(
+        email,
+        template_id,
+        get_personalization,
+        Settings.vanotify.services.va_gov.api_key,
+        { callback_metadata: { notification_type:, form_number:, statsd_tags: } }
+      )
+    end
+
+    def enqueue_email(at, template_id)
+      VANotify::EmailJob.perform_at(
+        at,
+        email,
+        template_id,
+        get_personalization,
+        Settings.vanotify.services.va_gov.api_key,
+        { callback_metadata: { notification_type:, form_number:, statsd_tags: } }
+      )
+    end
```

Comment:
The `send_email_now` and `enqueue_email` methods are nearly identical. If you're so inclined, this if/else logic can be handled in a more concise way. I'd love your thoughts:

```suggestion
def send(at: nil)
  template_id = TEMPLATE_IDS[notification_type]
  return unless template_id

  sent_to_va_notify = enqueue_or_send_email(template_id, at)
  StatsD.increment('silent_failure', tags: statsd_tags) if error_notification? && !sent_to_va_notify
end

private

...

def enqueue_or_send_email(template_id, at)
  args = [
    at,
    email,
    template_id,
    get_personalization,
    Settings.vanotify.services.va_gov.api_key,
    { callback_metadata: { notification_type:, form_number:, statsd_tags: } }
  ].compact

  job_method = at ? :perform_at : :perform_async
  VANotify::EmailJob.send(job_method, *args)
end
```

Reply:
As per our conversation in Slack, we decided not to go this route as using the `send` method may make it more difficult to determine which method is being used given the use case. The added verbosity is preferred for clarity of implementation.